src/Compilers/VisualBasic/Portable/Parser/Parser.vb

```diff
@@ -2945,16 +2945,22 @@ checkNullable:
                 Dim identifierNameOpt As IdentifierNameSyntax = Nothing
                 Dim asKeywordOpt As KeywordSyntax = Nothing
 
+                ' if there is a type character or As, then this must be a name
                 If CurrentToken.Kind = SyntaxKind.IdentifierToken AndAlso
-                        PeekNextToken().Kind = SyntaxKind.AsKeyword Then
+                        (DirectCast(CurrentToken, IdentifierTokenSyntax).TypeCharacter <> TypeCharacter.None OrElse
+                        PeekNextToken().Kind = SyntaxKind.AsKeyword) Then
 
                     identifierNameOpt = ParseIdentifierNameAllowingKeyword()
                     TryGetToken(SyntaxKind.AsKeyword, asKeywordOpt)
```

Comment:
I am not sure if this is going to properly handle ```V2%()``` as the input. For example ```Dim V2%()``` declares a field or a local named V2 with type Integer(). How is this going to be handled within a tuple type? Please add a test.